src/route-planning/cargoUnit.js

```diff
@@ -62,14 +71,36 @@ class CargoUnit extends OrbitingGameObject {
             linVel.normalize();
 
             this.timeInTransit += deltaTime;
-            this.distanceTraveled += this.lastVelocity.length() * deltaTime;            
-            
+            this.distanceTraveled += this.lastVelocity.length() * deltaTime;           
             this.rotation = Vector3.Cross(this.mesh.up, linVel);
+
+            this.captureRouteData();
+            this.encounterManager.update(deltaTime);
             this.physicsImpostor.applyImpulse(this.currentGravity.scale(deltaTime), this.mesh.getAbsolutePosition());
             this.currentGravity = Vector3.Zero();
         }
     }
 
+    captureRouteData() {
+        let node = new TransformNode("cargoNode", this.scene, true);
+        node.position = this.mesh.position.clone();
```

Comment:
instead of cloning use `copyFrom`. the vector3's are already there (and rotation quaternion should be created beforehand)